### DISCUSSION TITLE
Question about using Conda (and singularity) in SGE cluster

### DISCUSSION THREAD

**Discussion by tanyasarkjain** [nextflow-io/nextflow#5292] [Q&A]
Hi all,

I'm trying to use some Conda packages in my nextflow pipeline, and I'm having trouble getting Conda to work. Apologies if this question has been answered already - I looked at some passed issues on Git, and couldn't find someone with the same problem as me - I am new to using Conda with nextflow. 

I have tried a few different things.For one, in my nextflow.config file, under my processes section, I tried enabling Conda, via Conda.enabled = true, and then in my nextflow script, in the specific task that I wanted to use a Conda package, I put conda "bwa samtools hiphase" to specify a Conda environment with those packages, be created. I also tried, creating a Conda profile in my nextflow.config file, where I enabled Conda, and specified which packages to use.

```
`profiles {
  conda {
    conda.enabled = true
    process.conda = 'hiphase' 
  }

  apptainer {
    enabled = true
    autoMounts = true
  }
}

process {
  executor='sge'
  queue='long.q'
  clusterOptions = '-S /bin/bash'
  cache = true
}`
```

<img width="803" alt="Screenshot 2024-09-09 at 10 58 06 PM" src="https://github.com/user-attachments/assets/1386c183-310f-4cc0-90d5-c4b070eae061">

Would appreciate any insight for how to correctly go about this - Conda also works locally on the development node I am using.

**Comment by colindaven** [comment]
Here's some help from the nice new hello nextflow training docs on getting conda up and running:

https://training.nextflow.io/hello_nextflow/06_hello_config/#13-enable-conda-in-the-configuration-file

I used to use conda (now use singularity) and just installed envs from a yaml file on the command line, then specified that env in nextflow processes (so not even in the config file, but in the main.nf file) and had no issues.

**Comment by mribeirodantas** [comment]
Hi @tanyasarkjain ,

You're setting `conda.conda.enabled`, and adding the `conda` process directive to the conda scope. That's wrong. It should be instead:

```
conda {
  enabled = true
}

process {
  conda = 'hiphase'
}
```

Everything else should be the same. For other questions, please ask on https://community.seqera.io
